app/javascript/packs/utils/time_parser.js

```diff
@@ -0,0 +1,85 @@
+import log from "../utils/log";
+import moment from "moment-timezone";
+
+const TimeParser = () => {
+  return {
+    displayLocalDate(time, preciseTime) {
+      // Ensure we return if it's a big future day
+      if (preciseTime == null) {
+        preciseTime = false;
+      }
+      if (time < window.tomorrow) {
+        if (time > window.today) {
+          return time.format("h:mma");
+        } else if (time > window.yesterday) {
+          return `Yesterday ${time.format("h:mma")}`;
+        }
+      }
+      if (time.year() === moment().year()) {
+        if (preciseTime) {
+          return time.format("MMM Do[,] h:mma");
+        } else {
+          return time.format("MMM Do[,] ha");
+        }
+      } else {
+        if (preciseTime) {
+          return time.format("YYYY-MM-DD h:mma");
+        } else {
+          return time.format("YYYY-MM-DD");
+        }
+      }
+    },
+
+    preciseTimeSeconds(time) {
+      return time.format("YYYY-MM-DD h:mm:ss a");
+    },
+
+    localize() {
+      if (!window.timezone) {
+        window.timezone = moment.tz.guess();
+      }
+      moment.tz.setDefault(window.timezone);
+      window.yesterday = moment()
+        .subtract(1, "day")
+        .startOf("day");
+      window.today = moment().startOf("day");
+      window.tomorrow = moment().endOf("day");
+
+      let displayLocalDate = this.displayLocalDate;
+      let preciseTimeSeconds = this.preciseTimeSeconds;
+      // Write local time
+      $(".convertTime").each(function() {
+        let $this = $(this);
+        let text = $this.text().trim();
+        let time = null;
+        $this.removeClass("convertTime");
+        $this.addClass("convertedTime"); // So we can style it
+        if (!(text.length > 0)) {
+          return;
+        }
+        // If time is only a number, parse as a timestamp
+        // Otherwise, parse as ISO_8601 which is what convert_time strftimes into
+        if (/^\d+$/.test(text)) {
```

Comment:
Turns out I was wrong, or moment changed or something. It didn't actually parse timestamps prior to adding this.